packages/client/src/stores/notification.js

```diff
@@ -18,25 +16,27 @@ const createNotificationStore = () => {
     type = "info",
     icon,
     autoDismiss = true,
+    duration = 3000,
     count = 1
   ) => {
     if (block) {
       return
     }
 
-    // If peeking, pass notifications back to parent window
     if (get(routeStore).queryParams?.peek) {
       window.parent.postMessage({
         type: "notification",
         detail: {
           message,
           type,
           icon,
+          duration,
```

Comment:
We don't need to worry about checking for a valid value here, as the event is ultimately passed back to the parent window which runs through this same code, and will eventually only be used down in the `setTimeout` block below, where we do default it. Just explaining why I don't think we need to worry about making any changes here - this can stay as you have it.